backend/src/services/user.ts

```diff
@@ -42,4 +44,93 @@ async function getMenteeId(pairingId: string): Promise<string> {
   return pairing.menteeId;
 }
 
-export { getMentorId, getMenteeId };
+/**
+ * Updates a mentor entry
+ * @param updatedMentor - updated values for mentor
+ * @param userID - uid of mentor to update
+ * @returns Updated saved mentor
+ */
+async function updateMentor(updatedMentor: UpdateMentorRequestBodyType, userID: string) {
+  console.log("updatedMentor", updatedMentor);
+  const mentor = await Mentor.findById(userID);
+  if (!mentor) {
+    throw ServiceError.MENTOR_WAS_NOT_FOUND;
+  }
+  await validateCalendlyAccessToken(updatedMentor.personalAccessToken);
+  await validateCalendlyLink(updatedMentor.calendlyLink);
+  try {
+    if (mentor != null) {
+      mentor.name = updatedMentor.name;
```

Comment:
I would recommend using this mongoose [function](https://www.geeksforgeeks.org/mongoose-findbyidandupdate-function/) instead of hard coding the updates.

Reply:
Agreed, let's use mongoose functions here

for example, check LAK - https://github.com/TritonSE/LAK-Goods-Transport-Application/blob/6871096672726da034951abd18de7ab1c92aff8e/backend/services/user.js#L140